crates/bevy_debug_draw/src/debug_draw_3d.rs

```diff
@@ -0,0 +1,192 @@
+    // The mesh data is held as plain arrays here
+    // If we wish to extend to more than just lines we may need multiple pairs that will later be ass, e.g. vertices_line and vertices_triangle
+    vertices: Vec<[f32; 3]>,
+    colors: Vec<[f32; 4]>,
+    dirty: bool,
+    clear: bool,
+}
+
+impl Default for DebugDraw3D {
+    fn default() -> Self {
+        DebugDraw3D {
+            vertices: Default::default(),
+            colors: Default::default(),
+            dirty: true,
+            clear: true,
+        }
+    }
+}
+
+impl DebugDraw3D {
+    pub fn draw_line(&mut self, start: Vec3, end: Vec3, color: Color) {
+        self.vertices.push(start.into());
+        self.vertices.push(end.into());
+        self.colors.push(color.into());
+        self.colors.push(color.into());
+        self.set_dirty();
+    }
+
+    pub fn set_dirty(&mut self) {
+        self.dirty = true;
+    }
+
+    pub fn reset(&mut self) {
+        if self.clear {
+            self.vertices.clear();
+            self.colors.clear();
+        }
+        self.dirty = false;
+    }
+
+    pub fn set_clear(&mut self, clear: bool) {
+        self.clear = clear;
+    }
+}
+/// This component marks the internal entity that does the mesh drawing.
+#[derive(Default)]
+struct DebugDraw3DComponent;
+
+/// The Material holding the shader for debug drawing
+#[derive(RenderResources, Default, TypeUuid)]
+#[uuid = "188f0f97-60b2-476a-a749-7a0103adeeba"]
+pub struct DebugDraw3DMaterial;
+
+///This system sets up the entity holding the actual mesh for drawing as well as the render pipeline step for the shader.
+fn setup_debug_draw_3d(
+    mut commands: Commands,
+    mut shaders: ResMut<Assets<Shader>>,
+    mut materials: ResMut<Assets<DebugDraw3DMaterial>>,
+    mut render_graph: ResMut<RenderGraph>,
+) {
+    // Crate a shader Pipeline
+    let mut p = PipelineDescriptor::default_config(ShaderStages {
+        vertex: shaders.add(Shader::from_glsl(
+            ShaderStage::Vertex,
+            include_str!("shaders/debugDrawLine.vert"),
+        )),
+        fragment: Some(shaders.add(Shader::from_glsl(
+            ShaderStage::Fragment,
+            include_str!("shaders/debugDrawLine.frag"),
+        ))),
+    });
+    p.primitive.topology = PrimitiveTopology::LineList;
+    p.primitive.cull_mode = CullMode::None;
+
+    // add the material to the pipeline
+    render_graph.add_system_node(
+        "debug_draw_3d",
+        AssetRenderResourcesNode::<DebugDraw3DMaterial>::new(false),
+    );
+    // connect that node stage the MAIN_PASS node
+    render_graph
+        .add_node_edge("debug_draw_3d", base::node::MAIN_PASS)
+        .unwrap();
+
+    let material_instance = materials.add(DebugDraw3DMaterial {});
+
+    // Spawn a entity that will do the debug drawing with its mesh
+    commands
+        .spawn(MeshBundle::default())
+        .with(material_instance)
+        .with(DebugDraw3DComponent::default())
+        .current_entity()
+        .unwrap();
```

Comment:
```suggestion
    commands
        .spawn(MeshBundle::default())
        .with(material_instance)
        .with(DebugDraw3DComponent::default());
```
This can be removed, as you arecrurrently not doing anything with the entity.